src/sentry/digests/backends/redis.py

```diff
@@ -110,6 +111,7 @@ class RedisBackend(Backend):
     """
     def __init__(self, **options):
         self.cluster, options = get_cluster_from_options('SENTRY_DIGESTS_OPTIONS', options)
+        self.locks = LockManager(RedisLockBackend(self.cluster))
```

Comment:
Right now, the locks share the same cluster as the data, but aren't actually colocated with the data that they are locking since lock prefix is included in the key used for determining data distribution/routing. For instance, the timeline key might be `d:t:mail:p:1`, but the lock key will be `l:d:t:mail:p:1`, so these will likely end up on separate nodes.

It might make sense to ignore the prefix key when determining distribution but keep it as part of the key used for actual storage (to avoid colliding with the data it's actually locking.)